umm-spec-lib/src/cmr/umm_spec/field_update.clj

```diff
@@ -132,15 +132,19 @@
 
 (defmethod apply-umm-list-update :find-and-replace
   [update-type umm update-field update-value find-value]
-  (if (seq (get-in umm update-field))
-    (let [update-value (util/remove-nil-keys update-value)]
-      ;; For each entry in update-field, if we find it using the find params,
-      ;; completely replace with update value
-      (update-in umm update-field #(distinct (map (fn [x]
-                                                    (if (value-matches? find-value x)
-                                                      update-value
-                                                      x))
-                                                  %))))
+  (if (seq (get-in umm update-field)) 
+    (let [update-value (if (sequential? update-value)
+                         (map #(util/remove-nil-keys %) update-value)
+                         (util/remove-nil-keys update-value))
+          ;; For each entry in update-field, if we find it using the find params,
+          ;; completely replace with update value
+          updated-umm (update-in umm update-field #(flatten 
```

Comment:
will mapcat work here instead of flatten?

Reply:
```
          updated-umm (update-in umm update-field #(mapcat (fn [x]
                                                                                                         (if (value-matches? find-value x)
                                                                                                           update-value
                                                                                                           x))
                                                                                                      %)))]
```

Reply:
I got 22 failures after changing it to mapcat from bulk_update_test.clj.

Reply:
user=> (mapcat #(if % {:a 1}) [{:a 1} {:b 2}])
([:a 1] [:a 1])
user=> (flatten (map #(if % {:a 1}) [{:a 1} {:b 2}]))
({:a 1} {:a 1})
They are not the same.

Reply:
Since I replace one object with possibly an array of objects, update-field could become [object [objects]] and I need to flatten it to [object1 object2... objectn]

Reply:
so I fiddled around and found a way to do it without flatten.
If anyone else thinks you should replace it with this then you can otherwise I'm going to set it to approved.
```
          updated-umm (update-in umm update-field (fn [values]
                                                    (reduce (fn [coll x]
                                                              (if (value-matches? find-value x)
                                                                (if (sequential? update-value)
                                                                  (concat update-value coll)
                                                                  (conj coll update-value))
                                                                (conj coll x)))
                                                            []
                                                            values)))]
```